### ISSUE TITLE
Trouble installing on Debian testing due to dependency problems

### ISSUE THREAD

**Issue by avila** [Ulauncher/Ulauncher#719]
I am having some trouble with dependencies on Debian testing. I tried downloading [the .deb file](https://github.com/Ulauncher/Ulauncher/releases/download/5.10.0/ulauncher_5.10.0_all.deb) and installing with `dpg -i`, as in the following:

```sh
sudo dpkg -i ulauncher_5.10.0_all.deb 
sudo apt -f install
sudo dpkg -i ulauncher_5.10.0_all.deb 
```
I apologize If this is a duplicate of #702. In this case, feel free to promptly close this issue and I will follow the development of that one instead. 

### complete output 

```sh
~/Downloads/ulau $ sudo dpkg -i ulauncher_5.10.0_all.deb 
Selecting previously unselected package ulauncher.
(Reading database ... 256679 files and directories currently installed.)
Preparing to unpack ulauncher_5.10.0_all.deb ...
Unpacking ulauncher (5.10.0) ...
dpkg: dependency problems prevent configuration of ulauncher:
 ulauncher depends on python3-pyinotify; however:
  Package python3-pyinotify is not installed.
 ulauncher depends on libkeybinder-3.0-0; however:
  Package libkeybinder-3.0-0 is not installed.
 ulauncher depends on gir1.2-keybinder-3.0 (>= 0.3); however:
  Package gir1.2-keybinder-3.0 is not installed.
 ulauncher depends on gir1.2-appindicator3-0.1; however:
  Package gir1.2-appindicator3-0.1 is not installed.
 ulauncher depends on python3-distutils-extra; however:
  Package python3-distutils-extra is not installed.
 ulauncher depends on python3-levenshtein; however:
  Package python3-levenshtein is not installed.
 ulauncher depends on python3-websocket; however:
  Package python3-websocket is not installed.

dpkg: error processing package ulauncher (--install):
 dependency problems - leaving unconfigured
Processing triggers for mailcap (3.69) ...
Processing triggers for gnome-menus (3.36.0-1) ...
Processing triggers for desktop-file-utils (0.26-1) ...
Processing triggers for gnome-icon-theme (3.12.0-3) ...
Processing triggers for hicolor-icon-theme (0.17-2) ...
Errors were encountered while processing:
 ulauncher
~/Downloads/ulau [1] $ sudo apt -f install
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Correcting dependencies... Done
The following packages will be REMOVED:
  ulauncher
0 upgraded, 0 newly installed, 1 to remove and 0 not upgraded.
1 not fully installed or removed.
After this operation, 6.261 kB disk space will be freed.
Do you want to continue? [Y/n] y
(Reading database ... 256993 files and directories currently installed.)
Removing ulauncher (5.10.0) ...
Processing triggers for hicolor-icon-theme (0.17-2) ...
Processing triggers for gnome-menus (3.36.0-1) ...
Processing triggers for gnome-icon-theme (3.12.0-3) ...
Processing triggers for mailcap (3.69) ...
Processing triggers for desktop-file-utils (0.26-1) ...
~/Downloads/ulau $ sudo dpkg -i ulauncher_5.10.0_all.deb
Selecting previously unselected package ulauncher.
(Reading database ... 256679 files and directories currently installed.)
Preparing to unpack ulauncher_5.10.0_all.deb ...
Unpacking ulauncher (5.10.0) ...
dpkg: dependency problems prevent configuration of ulauncher:
 ulauncher depends on python3-pyinotify; however:
  Package python3-pyinotify is not installed.
 ulauncher depends on libkeybinder-3.0-0; however:
  Package libkeybinder-3.0-0 is not installed.
 ulauncher depends on gir1.2-keybinder-3.0 (>= 0.3); however:
  Package gir1.2-keybinder-3.0 is not installed.
 ulauncher depends on gir1.2-appindicator3-0.1; however:
  Package gir1.2-appindicator3-0.1 is not installed.
 ulauncher depends on python3-distutils-extra; however:
  Package python3-distutils-extra is not installed.
 ulauncher depends on python3-levenshtein; however:
  Package python3-levenshtein is not installed.
 ulauncher depends on python3-websocket; however:
  Package python3-websocket is not installed.

dpkg: error processing package ulauncher (--install):
 dependency problems - leaving unconfigured
Processing triggers for mailcap (3.69) ...
Processing triggers for gnome-menus (3.36.0-1) ...
Processing triggers for desktop-file-utils (0.26-1) ...
Processing triggers for gnome-icon-theme (3.12.0-3) ...
Processing triggers for hicolor-icon-theme (0.17-2) ...
Errors were encountered while processing:
 ulauncher
~/Downloads/ulau [1] $ 
```

**Comment by troycurtisjr** [comment]
shouldn't you do:
```
sudo apt install ./ulauncher_5.10.0_all.deb 
```
as described on https://ulauncher.io/#Download ?

`dpkg` doesn't install dependencies, but `apt` does. So your first and last command correctly mention the necessary dependent packages aren't installed, and you `apt` command doesn't reference the deb.  You need to combine them into the single command :smile:

**Comment by avila** [comment]
I also tried that, but didn't work either and got following error. I was then trying to following [this comment](https://github.com/Ulauncher/Ulauncher/issues/702#issuecomment-801183792). 

`sudo apt install ./ulauncher_5.10.0_all.deb -f` also does not work. 

```
sudo apt install ./ulauncher_5.10.0_all.deb -f
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Note, selecting 'ulauncher' instead of './ulauncher_5.10.0_all.deb'
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 ulauncher : Depends: gir1.2-appindicator3-0.1 but it is not installable
E: Unable to correct problems, you have held broken packages.
```

**Comment by gornostal** [comment]
Fixed the issue by making the appindicator dependency optional. Gnome doesn't have indicators enabled by default anyway.
Feel free to reopen if you see any issues.
